mesonbuild/dependencies/pkgconfig.py

```diff
@@ -32,64 +32,144 @@
     from .._typing import ImmutableListProtocol
     from ..build import EnvironmentVariables
 
-class PkgConfigDependency(ExternalDependency):
+class PkgConfigInterface:
+    '''Base class wrapping a pkg-config implementation'''
+
+    @staticmethod
+    def instance(silent: bool, env: Environment, for_machine: MachineChoice) -> T.Optional[PkgConfigInterface]:
+        impl = PkgConfigCLI(silent, env, for_machine)
+        if not impl.found():
+            return None
+        return impl
+
+    def found(self) -> bool:
+        '''Return whether pkg-config is supported'''
+        raise NotImplementedError
+
+    def version(self, name: str) -> T.Optional[str]:
+        '''Return module version or None if not found'''
+        raise NotImplementedError
+
+    def cflags(self, name: str, allow_system: bool = False,
+               define_variable: T.Optional['ImmutableListProtocol[str]'] = None) -> T.List[str]:
```

Comment:
```suggestion
               define_variable: T.Optional[ImmutableListProtocol[str]] = None) -> T.List[str]:
```
You don't need the quotes because annotations are imported from future. Same apply to other functions.

Reply:
done